File: lab1/java/concurrent/Find.java
public class Find {
    public static void main(String[] args) {
        if (args.length != 1) {
            System.out.println("Use: java Main <size>");
            System.exit(1);
        }

        try {
            int size = Integer.parseInt(args[0]);

            int[][] matrix = Matrix.generateMatrix(size);
		
    	   Thread maxThread = new Thread(new MaxValueThread(matrix));
	   Thread minThread = new Thread(new MinValueThread(matrix));
	   
	   maxThread.start();
	   minThread.start(); 

           // System.out.printf("Max value: %d\n", Matrix.Max(matrix)); // 1T
           // System.out.printf("Min value: %d\n", Matrix.Min(matrix)); // 2T
            
        } catch (NumberFormatException e) {
            System.out.println("The size of the matrix must be an integer.");
            System.exit(1);
        }

        
    }
}

File: python/concurrent/find.py
import threading
import time

from matrix import generate_matrix, max, min
import sys


if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Use: python3 find.py <size>")
        sys.exit(1)

    try:
        size = int(sys.argv[1])
    except ValueError:
        print("The size of the matrix must be an integer.")
        sys.exit(1)
    
    matrix = generate_matrix(size)
    
    threads = [] 

    for threads in threads: 
        t1 = threading.Thread(target=max, args=matrix)
        t2 = threading.Thread(target=min, args=matrix)
        threads.append(t1)
        threads.append(t2)
        t1.start()
        t2.start()
    print(f'Max value: {max(matrix)}')
    print(f'Min value: {min(matrix)}')
    
